Kernel/Devices/USB/USBPipe.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/Types.h>
+#include <Kernel/Devices/USB/USBDescriptors.h>
+#include <Kernel/VM/Region.h>
+
+namespace Kernel::USB {
+
+//
+// A pipe is the logical connection between a memory buffer on the PC (host) and
+// an endpoint on the device. In this implementation, the data buffer the pipe connects
+// to is the physical buffer created when a USBTransfer is allocated.
+//
+class USBPipe {
```

Comment:
Since this is already in the `Kernel::USB` namespace, maybe we can just call it `Pipe`?
Same comment for all other things that start with `USB` in this namespace :)